### PR TITLE
docs(logger): update child logger docs section and snippets

### PR DESCRIPTION
**Pull request by shdq**
## Description of your changes

This PR is intended to improve docs for child loggers (`createChild` method). It is the last part of improving the child logger implementation, unit tests, and documentation. In the first PR #1178 child Logger implementation was changed, and new tests were introduced, in this PR #1264 implementation was improved, and new tests were introduced that cover all attributes inherited from a parent logger.

### Related issues, RFCs

**Issue number:** #483 

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published*
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/core/logger.md [hunk 1]
```diff
@@ -390,8 +390,9 @@ The error will be logged with default key name `error`, but you can also pass yo
 
 ### Using multiple Logger instances across your code
 
-Logger supports quick instance cloning via the `createChild` method.
-This can be useful for example if you want to enable multiple Loggers with different logging levels in the same Lambda invocation.
+The `createChild` method of the Logger allows you to create a child instance of the Logger, which inherits all of the attributes from its parent. You have the option to override any of the settings that can be set through the constructor, including [the default settings](#utility-settings), any [persistent attributes](#appending-persistent-additional-log-keys-and-values), and [the log formatter](#custom-log-formatter-bring-your-own-formatter).
```
Comment (dreamorosi) [inline docs/core/logger.md:393]:
```suggestion
The `createChild` method allows you to create a child instance of the Logger, which inherits all of the attributes from its parent. You have the option to override any of the settings and attributes from the parent logger, including [the its settings](#utility-settings), any [persistent attributes](#appending-persistent-additional-log-keys-and-values), and [the log formatter](#custom-log-formatter-bring-your-own-formatter). Once a child logger is created, the logger and its parent will act as separate instances of the Logger class, and as such any change to one won't be applied to the other. 
```

File: docs/core/logger.md [hunk 2]
```diff
@@ -390,8 +390,9 @@ The error will be logged with default key name `error`, but you can also pass yo
...
+
+The following example shows how to activate multiple Loggers and inherit the service name and persistent attributes while specifying different logging levels within a single Lambda invocation. As the result, only ERROR logs with all the inherited attributes will be displayed in CloudWatch Logs from the child logger.
```
Comment (dreamorosi) [inline docs/core/logger.md:395]:
```suggestion
 The following example shows how to create multiple Loggers that share service name and persistent attributes while specifying different logging levels within a single Lambda invocation. As the result, only ERROR logs with all the inherited attributes will be displayed in CloudWatch Logs from the child logger, but all logs emitted will have the same service name and persistent attributes.
```

**Review by dreamorosi — COMMENTED** [review]
Thanks for the PR!

Minor suggestions/rewording on a couple of sentences.

**Comment by shdq** [comment]
> Once a child logger is created, the logger and its parent will act as separate instances of the Logger class, and as such any change to one won't be applied to the other. 

I like this sentence! Thanks for the suggestions!

**Review by dreamorosi — APPROVED** [review]
LGTM!

Thank you for the help on this 🎉

**Pull request merged**